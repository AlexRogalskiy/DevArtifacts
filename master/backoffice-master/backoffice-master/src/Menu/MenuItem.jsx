import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import {
  ListItem,
  ListItemIcon,
  ListItemText,
  withStyles,
} from '@material-ui/core'

const styles = {
  isDisabled: {
    opacity: 0.5,
    pointerEvents: 'none',
  },
}

const MenuItem = ({
  redirectTo,
  url,
  title,
  icon: Icon,
  isDisabled,
  classes,
}) => (
  <ListItem
    button
    onClick={() => redirectTo(url)}
    className={classNames({
        [classes.isDisabled]: isDisabled,
      })}
  >
    {Icon ? (
      <ListItemIcon>
        <Icon />
      </ListItemIcon>
      ) : null}
    <ListItemText primary={title} />
  </ListItem>
)

MenuItem.propTypes = {
  redirectTo: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  isDisabled: PropTypes.bool,
  icon: PropTypes.func,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

MenuItem.defaultProps = {
  isDisabled: false,
  icon: null,
}

export default withStyles(styles)(MenuItem)
