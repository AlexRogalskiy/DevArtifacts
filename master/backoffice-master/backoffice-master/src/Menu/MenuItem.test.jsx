import React from 'react'
import { mount } from 'enzyme'

import Icon from '@material-ui/icons/AccessAlarm'

import MenuItem from './MenuItem'

it('renders correctly', () => {
  const tree = mount(<MenuItem
    title="Title"
    url="/"
    redirectTo={() => {}}
  />)

  expect(tree).toMatchSnapshot()
})

it('renders correctly with icon', () => {
  const tree = mount(<MenuItem
    title="Title"
    url="/"
    redirectTo={() => { }}
    icon={Icon}
  />)

  expect(tree).toMatchSnapshot()
})

it('renders correctly disabled', () => {
  const tree = mount(<MenuItem
    title="Title"
    url="/"
    redirectTo={() => {}}
    isDisabled
  />)

  expect(tree).toMatchSnapshot()
})
