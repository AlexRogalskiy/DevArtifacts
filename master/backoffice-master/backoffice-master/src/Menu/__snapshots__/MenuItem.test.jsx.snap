// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiButtonBase-root-13 MuiListItem-root-2 MuiListItem-default-5 MuiListItem-gutters-9 MuiListItem-button-10"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="MuiListItemText-root-16"
  >
    <span
      className="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
    >
      Title
    </span>
  </div>
  <span
    className="MuiTouchRipple-root-47"
  />
</div>
`;

exports[`renders correctly disabled 1`] = `
<div
  className="MuiButtonBase-root-13 MuiListItem-root-2 MuiListItem-default-5 MuiListItem-gutters-9 MuiListItem-button-10 MenuItem-isDisabled-1"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="MuiListItemText-root-16"
  >
    <span
      className="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
    >
      Title
    </span>
  </div>
  <span
    className="MuiTouchRipple-root-47"
  />
</div>
`;

exports[`renders correctly with icon 1`] = `
<div
  className="MuiButtonBase-root-13 MuiListItem-root-2 MuiListItem-default-5 MuiListItem-gutters-9 MuiListItem-button-10"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-55 MuiListItemIcon-root-54"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
    />
  </svg>
  <div
    className="MuiListItemText-root-16"
  >
    <span
      className="MuiTypography-root-22 MuiTypography-subheading-29 MuiListItemText-primary-19"
    >
      Title
    </span>
  </div>
  <span
    className="MuiTouchRipple-root-47"
  />
</div>
`;
